Created new App component

codingshibe opened the pull request:
Created Header component
Created ProductListItem component
Created ProductList component
Defined method in ProductList named getProducts to make a get call to server
Defined componentDidMount method to handle the page after it has initially loaded
Began initial styling with Bootstrap4 and CSS
![Screen Shot 2020-01-13 at 5 37 17 PM](https://user-images.githubusercontent.com/32403215/72306498-224aa680-362c-11ea-9077-1fc30ad5268e.png)

codingshibe committed:
Created new App component
Created Header component
Created ProductListItem component
Created ProductList component
Defined method in ProductList named getProducts to make a get call to server
Defined componentDidMount method to handle the page after it has initially loaded
Began initial styling with Bootstrap4 and CSS

thebearingedge commented on client/components/ProductList.jsx:
```diff
@@ -0,0 +1,50 @@
+  }
+
+  displayProducts() {
+    const productList = [...this.state.products];
```
This isn't necessary as `map` does not mutate the original array.

thebearingedge commented on client/components/ProductList.jsx:
```diff
@@ -0,0 +1,50 @@
+    super(props);
+    this.state = { products: [] };
+    this.getProducts = this.getProducts.bind(this);
+    this.displayProducts = this.displayProducts.bind(this);
```
You shouldn't need to bind these.

codingshibe committed:
Removed the bind methods from the constructor
Changed the map method to map over this.state.products
Removed productList variable containing spread operator
